tests/csm/rest/test_iam_users.py

```diff
@@ -4638,3 +4651,350 @@ def test_42272(self):
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.CREATED,
+                                   "Create IAM User failed")
+        user_id = resp['tenant'] + "$" + resp['user_id']
+        self.log.info("Step 2: Delete IAM User")
+        resp = self.csm_obj.delete_iam_user(user_id, purge_data=True)
+        self.log.debug("Verify Response : %s", resp)
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.OK,
+                                   "Delete IAM User failed")
+        self.log.info("Step 3: List IAM users with delete user name as marker")
+        resp = self.csm_obj.list_iam_users_rgw(marker=resp['user_id'])
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.OK,
+                                 "List IAM User failed")
+        assert_utils.assert_equals(len(resp["keys"]), 0, "Users list is not empty")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-42284')
+    def test_42284(self):
+        """
+        Test GET IAM user with marker in between the displayed list.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step 1: Creating %s IAM users.", self.csm_conf["common"]["num_users"])
+        users_list = []
+        for count in range(self.csm_conf["common"]["num_users"]):
+            resp = self.csm_obj.verify_create_iam_user_rgw(verify_response=True)
+            assert_utils.assert_true(resp[0], resp[1])
+            users_list.append(resp[1]["user_id"])
+            self.log.info("%s IAM user created", count + 1)
+        self.log.info("Created users: %s", users_list)
+        self.created_iam_users = users_list
+
+        self.log.info("Step 2: Send GET request with max_entries as 5")
+        resp = self.csm_obj.list_iam_users_rgw(
+                                max_entries=self.csm_conf["common"]["max_entries"])
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.OK, "Status check failed")
+        resp_dict = resp.json()
+        get_user_list = resp_dict["users"]
+        count = resp_dict["count"]
+        assert_utils.assert_equals(count, self.csm_conf["common"]["max_entries"], 
+                                 "Entries not returned as expected")
+        user_index = self.csm_obj.random_gen.randrange(1, count)
+        marker = get_user_list[user_index]
+
+        self.log.info("Step 3: Send GET request with max_entries as 15 and "
+                      "marker as in between user")
+        resp = self.csm_obj.list_iam_users_rgw(max_entries=15, marker=marker)
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.OK, "Status check failed")
+        count = resp_dict["count"]
+        assert_utils.assert_equals(count, 15, "Entries not returned as expected")
```

Comment:
This may not be 15 as we have only created 10. So it should be 10 + Internal user

Reply:
Okay

Reply:
done